src/app/qgsdatumtransformtablewidget.cpp

```diff
@@ -0,0 +1,265 @@
+/***************************************************************************
+    qgsdatumtransformtablewidget.cpp
+     --------------------------------------
+    Date                 : 28.11.2017
+    Copyright            : (C) 2017 Denis Rouzaud
+    Email                : denis.rouzaud@gmail.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsdatumtransformtablewidget.h"
+
+#include "qgscoordinatetransform.h"
+#include "qgsdatumtransformdialog.h"
+
+
+QgsDatumTransformTableModel::QgsDatumTransformTableModel( QObject *parent )
+  : QAbstractTableModel( parent )
+{
+}
+
+void QgsDatumTransformTableModel::setTransformContext( const QgsCoordinateTransformContext &context )
+{
+  mTransformContext = context;
+  reset();
+}
+
+void QgsDatumTransformTableModel::removeTransform( const QModelIndexList &indexes )
+{
+  QgsCoordinateReferenceSystem sourceCrs;
+  QgsCoordinateReferenceSystem destinationCrs;
+  for ( QModelIndexList::const_iterator it = indexes.constBegin(); it != indexes.constEnd(); it ++ )
+  {
+    if ( it->column() == SourceCrsColumn )
+    {
+      sourceCrs = QgsCoordinateReferenceSystem( data( *it, Qt::DisplayRole ).toString() );
+    }
+    if ( it->column() == DestinationCrsColumn )
+    {
+      destinationCrs = QgsCoordinateReferenceSystem( data( *it, Qt::DisplayRole ).toString() );
+    }
+    if ( sourceCrs.isValid() && destinationCrs.isValid() )
+    {
+      mTransformContext.removeSourceDestinationDatumTransform( sourceCrs, destinationCrs );
+      reset();
+      break;
+    }
+  }
+}
+
+
+int QgsDatumTransformTableModel::rowCount( const QModelIndex &parent ) const
+{
+  Q_UNUSED( parent );
+  return mTransformContext.sourceDestinationDatumTransforms().count()
+#ifdef singlesourcedest
```

Comment:
How is this controlled, why is this used?

Reply:
all the code commented out is left from first implementation where we thought we would need several containers for transforms (source only, destination only, source & destination).
After some thoughts, I think source & destination is the only required container:
- transforms are reversible: it makes no sense to distinguish source and destination
- I don't think there is a practical use case for source or destination only transform (i.e. going to WGS84 only). 
But I'm not 100% sure for the latter, I decided to keep the code commented in case the reviewer -- you ;) -- thinks it's better to bring that back in.